proto/backend.proto

```diff
@@ -306,3 +306,160 @@ message StreamPacket {
   // JSON-encoded data to publish into a channel.
   bytes data = 1;
 }
+
+//-----------------------------------------------------------------
+// Storage -- EXPERIMENTAL and is subject to change until 12.??
+//-----------------------------------------------------------------
+
+service Storage {
+  // Check if the instance settings can be saved.  This is similar to a standard 
+  // mutating addmission hook, except that the properties are already typed protobuf
+  rpc MutateInstanceSettings(InstanceSettingsAdmissionRequest) returns (InstanceSettingsResponse);
```

Comment:
How long do you expect this RPC to live? Is it transitional until instance settings are implemented as a platform resource?

Reply:
Good question --- an alternative would be to shove it into the `object_bytes` for the more k8s friendly style below 🤔 

I added this native one so we can work with typed values, but that may force us to live with this less than great structure for longer than we want!

Reply:
I updated this -- I think we can shove the `AppInstanceSettings` + `DataSourceInstanceSettings` into bytes and use the same hook in the initial versions.